lib/lotus/mailer/rendering.rb

```diff
@@ -4,17 +4,35 @@ module Mailer
     #
     # @since 0.1.0
     module Rendering
-
-      # Render a single template with the specified format.
-      #
-      # @return [String] the output of the rendering process.
-      #
-      # @since 0.1.0
-      def render(format)
-        templates
-        @templates[format].render
+      def self.extended(base)
+        base.class_eval do
+          include InstanceMethods
+        end
       end
 
+      module InstanceMethods
+        # Initialize a mailer
+        #
+        # @param locals [Hash] a set of objects available during the rendering process.
+        #
+        # @since 0.1.0
+        def initialize(locals = {})
+          @locals   = locals
+          @scope    = self
+        end
+
+        # Render a single template with the specified format.
+        #
+        # @param format [symbol] template's format
+        #
+        # @return [String] the output of the rendering process.
+        #
+        # @since 0.1.0
+        def render(format)
+          self.class.templates
```

Comment:
As we've said before, if we freeze everything in one test then the rest don't pass, we get the message that everything is frozen. So yes, we have a problem with the tests and the lazyness. We've had it since the second issue but since no one could help us we did the rest ignoring that part.